lib/service/service.go

```diff
@@ -397,6 +397,11 @@ func newTeleportProcess(cfg *Config) (Process, error) {
 // Run starts teleport processes, waits for signals
 // and handles internal process reloads.
 func Run(ctx context.Context, cfg Config, newTeleport NewProcess) error {
+	// GRPC client logs are very verbose,
+	// so only turn them on with any level only in debug mode
+	if cfg.Debug {
+		setDebugGRPCLogger(log.Logger.WriterLevel(logrus.DebugLevel))
```

Comment:
This looks suspect as implicit configuration/functionality might be extremely confusing and hard to get rid of. Would it make sense to configure the logger at the call-site instead of here as Run might be used in another setting where this might have an unexpected effect?

Reply:
not sure what do you mean by that, an you clarify?

Reply:
Yes, so my concerns are:
  * it looks out of place since it is not critical to the Run's operation
  * I've ran into a similar setup [here](https://github.com/gravitational/teleport/blob/3e3495c481798677d89e160f635d313d6ceb56d6/lib/utils/cli.go#L152-L154) which causes [this](https://github.com/gravitational/teleport/issues/3205) - it basically made the function non-applicable in certain situations and hard to opt-out of this behavior
  * if the gRPC logging has been turned off explicitly for some reason at some point, invoking Run would suddenly revert the switch and it might be hard to spot when troubleshooting

Reply:
Got it, where would you place it instead? I have to place it before any GRPC calls, but after init and only on debug mode.